cla_public/apps/checker/validators.py

```diff
@@ -73,21 +73,41 @@ def __call__(self, form, field):
         if len(field.data) < 1:
             message = self.message
             if message is None:
-                message = field.gettext("Must select at least one option.")
+                message = field.gettext("Select at least one option")
             raise ValidationError(message)
 
 
 class MoneyIntervalAmountRequired(object):
-    def __init__(self, message=None):
+    def __init__(self, message=None, freq_message=None, amount_message=None):
         self.message = message
+        self.freq_message = freq_message
+        self.amount_message = amount_message
 
     def __call__(self, form, field):
         amount = field.form.per_interval_value
+        interval = field.form.interval_period
+        amount_not_set = amount.data is None
+        nonzero_amount = amount.data > 0
+        interval_selected = interval.data != ""
 
-        if not amount.errors and amount.data is None:
-            message = self.message or field.gettext(u"Please provide an amount")
+        if not amount.errors and interval_selected and amount_not_set and self.amount_message:
```

Comment:
This whole method is extremely hard to read now. In particular, I would be interested in making this `if` statement not have four conditions (either by combining them into a variable first or -- probably preferably -- moving one of more other `if` statements before it so that we're never dealing with so many factors all at once (eg if we can do a particular thing in all cases where one particular thing is true, then we might be able to do that early on and then not have to include it in any other cases).

We could also maybe split that intervals stuff out into a method, which would at least make that pathway less intimidating to take in.

Happy for you to work on this yourself if you'd like to, more than happy to pair on it.